### ISSUE TITLE
Global styles created with createGlobalStyle don't update on unmount

### ISSUE THREAD

**Issue by PontusHorn** [styled-components/styled-components#2769]
This issue occurs when a component created with `createGlobalStyle` is dynamic (e.g. through props) and more than one instance of the component may be mounted simultaneously.

What seems to happen, as far as I understand it, is this:

1. Instance 1 is mounted, injecting the appropriate CSS
2. Instance 2 is mounted, updating the CSS in question to reflect its own props
3. Instance 2 is unmounted, but the CSS remains unchanged.

### System:
 - OS: macOS Mojave 10.14.6
 - CPU: (8) x64 Intel(R) Core(TM) i7-8569U CPU @ 2.80GHz
 - Memory: 123.61 MB / 16.00 GB
 - Shell: 3.0.2 - /usr/local/bin/fish
### Binaries:
 - Node: 10.16.3 - ~/.config/nvm/10.16.3/bin/node
 - Yarn: 1.17.3 - ~/.config/nvm/10.16.3/bin/yarn
 - npm: 6.9.0 - ~/.config/nvm/10.16.3/bin/npm
### npmPackages:
 - babel-plugin-styled-components: ^1.10.0 => 1.10.1
 - styled-components: ^5.0.0-beta.9 => 5.0.0-beta.8-groupsizefix

## Reproduction

https://codesandbox.io/embed/style-with-styled-components-kvsrk

## Steps to reproduce

In the codesandbox above, check the checkbox and uncheck it again.

## Expected Behavior

When the checkbox is unchecked, the background should go back to white.

## Actual Behavior

The background remains yellow as the styles from the unmounted component are kept around.

**Comment by Adidi** [comment]
I think the fact the you have 2 `<Background />` components mess up your style.
If you remove the top one (without color prop) it will work !

**Comment by PontusHorn** [comment]
Thanks, but the reproduction isn't indicative of my real use case. The first `<Background />` is there to demonstrate the problem.

In my actual use case, the workaround was to not use props inside components created with `createGlobalStyle` so that it could handle multiple instances being mounted/unmounted across different areas of the app.

**Comment by Ketler13** [comment]
Could anyone help with this problem?
Please take a look at https://github.com/styled-components/styled-components/issues/2979#issuecomment-579123433

Incrementing instance counter possibly leads to a memory leak in SSR.

We have SSR with styled components. When server has started for a few days, such error appeared (top of the stack where styled components are mentioned):

RangeError: Invalid typed array length: -2147483648
at new Uint32Array ()
at DefaultGroupedTag.insertRules (/usr/src/app/node_modules/styled-components/dist/styled-components.cjs.js:417:25)
at StyleSheet.insertRules (/usr/src/app/node_modules/styled-components/dist/styled-components.cjs.js:654:19)
at GlobalStyle.createStyles (/usr/src/app/node_modules/styled-components/dist/styled-components.cjs.js:1644:16)
at GlobalStyle.renderStyles (/usr/src/app/node_modules/styled-components/dist/styled-components.cjs.js:1655:10)
at GlobalStyleComponent (/usr/src/app/node_modules/styled-components/dist/styled-components.cjs.js:1690:19)

This happens when page is rendered. I checked source code and in https://github.com/styled-components/styled-components/blob/147b0e9a1f10786551b13fd27452fcd5c678d5e0/packages/styled-components/src/sheet/GroupedTag.js there is a insertRules method.
There is a line where new size for groupSizes is calculated:

const newSize = BASE_SIZE << ((group / BASE_SIZE) | 0);

Error described above happens when BASE_SIZE << 23 ,e.g. group === 5888.
Argument group in this method comes from other module - https://github.com/styled-components/styled-components/blob/147b0e9a1f10786551b13fd27452fcd5c678d5e0/packages/styled-components/src/sheet/GroupIDAllocator.js . This is a plain counter that increments every time new group is created.

When group equals 5632, bitwise operation from above equals this BASE_SIZE << 22 === 1073741824. And new Uint32Array is created with size of 1073741824.

Does someone experience the same issue? To reproduce this problem, please, start SSR server and try to access your application via curl in a loop 1000, 2000 or even more and log to console newSize variable.

**Comment by mrmarktyy** [comment]
This Range Error seems still happening for me in production with styled-components v5.0.0 

```
RangeError: Invalid typed array length: -2147483648
 at new Uint32Array (<anonymous>)
 at DefaultGroupedTag.insertRules (/app/node_modules/styled-components/dist/styled-components.cjs.js:335:25)
 at StyleSheet.insertRules (/app/node_modules/styled-components/dist/styled-components.cjs.js:579:19)
 at ComponentStyle.generateAndInjectStyles (/app/node_modules/styled-components/dist/styled-components.cjs.js:1290:20)
 at useInjectedStyle (/app/node_modules/styled-components/dist/styled-components.cjs.js:1463:120)
 at useStyledComponentImpl (/app/node_modules/styled-components/dist/styled-components.cjs.js:1490:28)
```

Here is stack trace I can find.

**Comment by probablyup** [comment]
Try styled-components@test

We have a patch update coming soon that should take care of it

**Comment by mrmarktyy** [comment]
Thanks @probablyup 

We noticed v5.0.1 was released recently, so we have bumped to v5.0.1
unfortunately, It looked like the problem still existed.

Error stack trace was also the same as above.
